Frameworks/UIKit/UIImage.mm

```diff
@@ -382,7 +382,7 @@ - (instancetype)initWithContentsOfFile:(NSString*)pathAddr {
 
         if (EbrAccess(pathStr, 0) == -1) {
             id pathFind =
-                [bundle pathForResource:[NSString stringWithCString:newStr] ofType:nil inDirectory:nil forLocalization:@"English"];
+                [bundle pathForResource:[NSString stringWithCString:newStr] ofType:@"png" inDirectory:nil forLocalization:@"English"];
```

Comment:
This seems really strange. If I read this right, we test to see if 1) there's an extension separator, 2) we're a retina device, then tack on an `@2x`, and if that file isn't found we try again by tacking on a .png extension via pathForResource. If that isn't found, we try to access the file verbatim, and if that isn't found, we try again by tacking on a .png. 

I can think of some situations where it does what you don't want. For instance it appears it will never return an image when given a full _bundle_ path. (ie, `[UIImage initWithContentsOfFile:@"abc.jpg"]` where abc.jpg is only part of en.lproj. It seems like it'll look in the bundle first for abc@2x.jpg.png and then abc.jpg.png and of course never find it). And if the intent is to return a bundled png given a partial path, it also seems like it'll never return an `@2x` image. (ie, if this is to enable `[UIImage initWithContentsOfFile:@"abc"]` it will never return abc@2x.png, since there's no extension separator). 

Can you explain a little more the motivation behind this change? I'd be happy to give suggestions. :)

Dave